File: src/Day08_DictionariesAndMaps.java
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class Day08_DictionariesAndMaps {
    public static void main(String []argh){

        Scanner in = new Scanner(System.in);
        int n = in.nextInt();

        Map<String,Integer > phoneNumber = new HashMap<>();

        for(int i = 0; i < n; i++){
            String name = in.next();
            int phone = in.nextInt();
            phoneNumber.put(name,phone) ;
        }
        while(in.hasNext()){
            String s = in.next();

            if(phoneNumber.containsKey(s)){
                Integer no=phoneNumber.get(s);
                System.out.println(s+"=" + no);
            }
            else {
                System.out.println("Not found");
            }
        }
        in.close();
    }
}

File: src/Day08_DictionariesAndMaps.kt
import java.io.PrintWriter
import java.util.*
import kotlin.math.pow

/**
 * Tutorials > 30 Days of Code > Day 8: Dictionaries and Maps
 *
 * https://www.hackerrank.com/challenges/30-dictionaries-and-maps
 * @author lizarragadev
 *
 */

val map = mutableMapOf<String, String>()

fun main(args: Array<String>) {
    _writer.day08()
    _writer.flush()
}

fun PrintWriter.day08() {
    val n = readInt()
    if(n in 1..10.toDouble().pow(5.toDouble()).toInt()) {
        fillMap08(n)
        val sc = Scanner(System.`in`)

        while(!sc.hasNext()) {
            val query = sc.next()
                if(query in map)
                    println("$query=${map[query]}")
                else
                    println("Not found")
        }
    }
}

fun fillMap08(n: Int){
    for (i in 1..n) {
        val key = read()
        val value = read()
        if(!key.isNullOrEmpty() && value.length == 8) {
            map[key] = value
        }
    }
}